### PR TITLE
fix #71 init of core.Grid metadata

### LINKED ISSUE #71: core.Grid metadata badly initialized
instance variable metadata is shared between instances of Grid()

### PR DESCRIPTION
**Pull request by ReneHamburger1993**
fix for #71  prevent sharing of mutable dict between instances of core.Grid() class

### CODE REVIEW

File: gridData/core.py [hunk 1]
```diff
@@ -131,7 +131,10 @@ def __init__(self, grid=None, edges=None, origin=None, delta=None,
             'PYTHON': self._load_python,  # compatibility
         }
 
-        self.metadata = metadata
+        if metadata is not None:
```
Comment (orbeckst) [inline gridData/core.py]: If you want to you can also write it as
```python
self.metadata = metadata if metadata is not None else {}
```

File: gridData/core.py [hunk 2]
```diff
@@ -131,7 +131,10 @@ def __init__(self, grid=None, edges=None, origin=None, delta=None,
...
+            self.metadata = metadata
+        else:
+            self.metadata = {} # prevent sharing of mutable dict between instances
```
Comment (orbeckst) [inline gridData/core.py]: No need for the comment ;-)

File: gridData/core.py (hunk 1 above)
Comment (ReneHamburger1993) [reply gridData/core.py]: done

**Comment by orbeckst** [comment]
Thank you @ReneHamburger1993 !

**Pull request merged**